torch/_dynamo/compiled_autograd.py

```diff
@@ -0,0 +1,176 @@
+                    val,
+                    self.source("sizes", idx),
+                    dynamic_dim=DimDynamic.DYNAMIC,
+                ),
+                hint=val,
+            )
+            for idx, val in enumerate(sizes)
+        ]
+        self.bind_tensors_to_proxies(sizes, sizes_proxy)
+
+        # TODO(jansel): are all these modes needed?
+        self.stack.enter_context(decompose({}))
+        self.stack.enter_context(self.fake_tensor_mode)
+        self.stack.enter_context(self.proxy_mode.sym_mode)
+        self.stack.enter_context(self.proxy_mode)
+        self.stack.enter_context(disable_autocast_cache())
+        self.stack.enter_context(disable_proxy_modes_tracing(enable_current=True))
+        return inputs, sizes
+
+    def tensor_pre_hook(self, inputs, hook_id, i: int):
+        hook = self.hooks_proxy[hook_id]
+        proxy = self.fx_tracer.create_proxy(
+            "call_function",
+            call_hook,
+            (
+                hook,
+                self.to_proxy(inputs[i]),
+            ),
+            {},
+        )
+        with disable_proxy_modes_tracing():
+            inputs[i] = inputs[i].clone()
+            self.bind_tensors_to_proxies([inputs[i]], [proxy])
+        return inputs
+
+    def pre_hook(self, inputs, hook_id):
+        hook = self.hooks_proxy[hook_id]
+        proxies = self.fx_tracer.create_proxy(
+            "call_function",
+            call_hook,
+            (
+                hook,
+                self.to_proxy(inputs),
+            ),
+            {},
+        )
+        with disable_proxy_modes_tracing():
+            inputs = [x.clone() for x in inputs]
+            self.bind_tensors_to_proxies(inputs, proxies)
+        return inputs
+
+    def post_hook(self, outputs, inputs, hook_id):
+        hook = self.hooks_proxy[hook_id]
+        proxies = self.fx_tracer.create_proxy(
+            "call_function",
+            call_hook,
+            (
+                hook,
+                self.to_proxy(outputs),
+                self.to_proxy(inputs),
+            ),
+            {},
+        )
+        with disable_proxy_modes_tracing():
+            outputs = [x.clone() for x in outputs]
+            self.bind_tensors_to_proxies(outputs, proxies)
+        return outputs
+
+    def end_capture(self, outputs):
+        self.stack.close()
+        self.fx_tracer.create_node(
+            "output",
+            "output",
+            (self.fx_tracer.create_arg(self.to_proxy(outputs)),),
+            {},
+        )
+        return self.compiler_fn(
+            GraphModule(self.fx_tracer.root, self.fx_tracer.graph, "CompiledAutograd")
+        )
+
+    def to_proxy(self, t):
+        if t is None:
+            return None
+        if isinstance(t, list):
+            return [self.to_proxy(x) for x in t]
+        if isinstance(t, tuple):
+            return tuple(self.to_proxy(x) for x in t)
+        return fetch_tensor_proxy(self.fx_tracer)(t).proxy
```

Comment:
nit - iirc the get_proxy_slot error (under the hood here) is really weird for unexpected types, consider a try/catch to give it a clearer error. No strong opinion.

Reply:
There will only be tensors/symints here by construction.

Reply:
assert? otherwise sgtm